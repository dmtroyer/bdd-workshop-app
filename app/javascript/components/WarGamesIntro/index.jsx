import React from 'react';
import { Jumbotron, Button } from 'reactstrap';
import { Link } from 'react-router-dom';
import Typist from 'react-typist';
import MainContainer from '../MainContainer';

const WarGamesIntro = () => (
  <MainContainer dataTestId="home">
    <h1>Play with me?</h1>
    <Link className="jumbotron-link" to="/game" data-testid="game-start">
      <Jumbotron data-testid="wargames" className="m-3">
        <div className="p-3 wargames-dotty">
          <Typist>
            Games make mistakes.
            <div className="question">SHALL WE PLAY A GAME?</div>
          </Typist>
        </div>
        <div className="pt-3">
          <Button color="primary" className="float-right">
            Play Game
          </Button>
        </div>
      </Jumbotron>
    </Link>
  </MainContainer>
);
export default WarGamesIntro;
