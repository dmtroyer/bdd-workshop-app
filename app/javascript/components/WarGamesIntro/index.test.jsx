import React from 'react';
import { shallow } from 'enzyme';
import WarGamesIntro from '.';

describe('WarGamesIntro', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<WarGamesIntro />);
    expect(wrapper).toMatchSnapshot();
  });
});
