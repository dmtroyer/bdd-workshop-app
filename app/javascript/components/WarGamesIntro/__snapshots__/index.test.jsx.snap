// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarGamesIntro renders correctly 1`] = `
<MainContainer
  dataTestId="home"
>
  <h1>
    Play with me?
  </h1>
  <Link
    className="jumbotron-link"
    data-testid="game-start"
    replace={false}
    to="/game"
  >
    <Jumbotron
      className="m-3"
      data-testid="wargames"
      tag="div"
    >
      <div
        className="p-3 wargames-dotty"
      >
        <Typist
          avgTypingDelay={70}
          className=""
          cursor={Object {}}
          delayGenerator={[Function]}
          onCharacterTyped={[Function]}
          onLineTyped={[Function]}
          onTypingDone={[Function]}
          startDelay={0}
          stdTypingDelay={25}
        >
          Games make mistakes.
          <div
            className="question"
          >
            SHALL WE PLAY A GAME?
          </div>
        </Typist>
      </div>
      <div
        className="pt-3"
      >
        <Button
          className="float-right"
          color="primary"
          tag="button"
        >
          Play Game
        </Button>
      </div>
    </Jumbotron>
  </Link>
</MainContainer>
`;
